File: Transport/JS/Transport/view/tire/TireRemoveReason.js
T.view.TireRemoveReason = Ext.extend(Kdn.view.BaseGrid, {
    requireModels: 'TireTechState',
    modelName: 'TireRemoveReason',
    constructor: function(cfg) {
        cfg = cfg || {};
        Ext.apply(cfg, {

            colModel: new Ext.grid.ColumnModel({
               defaults:{filter:{}},
                columns: [
                    {
                        dataIndex: 'TireRemoveReasonId',
                        header: 'Код',
                        width: 130,
                        editor: { xtype: 'kdn.editor.id' }
                    },
                    {
                        dataIndex: 'TireRemoveReasonName',
                        header: 'Наименование причины снятия шины',
                        width: 450,
                        editor: { xtype: 'kdn.editor.textfield' }
                    },
                    {
                        dataIndex: 'isWriteOff',
                        header: 'Списание шины',
                        width: 200,
                        renderer:function(v) {
                            if (!Ext.isBoolean(v)) return v;
                            return (!!v) ? 'Да' : 'Нет';
                        },
                        editor : {
                           xtype:'kdn.editor.booleanfield'                           
                        }
                    },
                    {
                        dataIndex: 'State',
                        header: 'Тех. состояние',
                        renderer:Kdn.Renderer.object("TireTechStateName"),
                        width: 200,
                        editor: { xtype: 'combo.tiretechstate', allowBlank:true }  
                    }
                ]
            })
        });

        T.view.TireRemoveReason.superclass.constructor.call(this, cfg);
    }
});

Ext.reg('view.tireremovereason', T.view.TireRemoveReason);

File: Transport/Models/tire/TireRemoveReason.cs
using Castle.ActiveRecord;
using Kdn.Ext.Attributes;

namespace Transport.Models.tire
{
    [Model,ActiveRecord]
    public class TireRemoveReason:ActiveRecordBase<TireRemoveReason>
    {
        [IdProperty, PrimaryKey]
        public int TireRemoveReasonId { get; set; }
        [Property]
        public string TireRemoveReasonName { get; set; }
        [Property]
        public bool isWriteOff { get; set; }

        [BelongsTo("TireTechStateId"), AllowBlank]
        public TireTechState State { get; set; }

    }
}
